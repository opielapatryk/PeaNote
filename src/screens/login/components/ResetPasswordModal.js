import React,{useState} from 'react';
import {View, Modal,Text,TextInput,Pressable,} from 'react-native';
import auth from '@react-native-firebase/auth'
import { styles } from '../../../../assets/styles/styles';

const ResetPasswordModal = ({modalVisible,setModalVisible}) => {
    const [resetEmail,setResetEmail] = useState("")
    const [resetPasswordLog,setResetPasswordLog] = useState("Email")

    const checkIfEmailExist = () => {
        auth().sendPasswordResetEmail(resetEmail)
        .then(() => {
          setResetPasswordLog('Check your mailbox!')
          setResetEmail('')
        })
        .catch(() => {
          setResetPasswordLog('Email address is badly formatted.')
          setResetEmail('')
        });
      }

    return (
            <Modal animationType="slide" transparent={true} visible={modalVisible}>
                <View style={styles.modalPasswordResetView}>
                <View style={styles.modalPasswordResetViewChild}>
                    <Text style={styles.modalPasswordResetHeader}>Where should we send a reset link?</Text>
                    <Text style={styles.modalPasswordResetParagraph}>Enter the email associated with your account to reset your password.</Text>
                    <TextInput style={styles.modalPasswordResetTextInput} placeholder={resetPasswordLog} value={resetEmail} onChangeText={text=>setResetEmail(text)}/>
                    <View style={{flexDirection:'row',justifyContent:'space-between'}}>
                    <Pressable onPress={()=>checkIfEmailExist()} style={styles.modalPasswordResetButtonNext}><Text style={styles.modalPasswordResetButtonNextText}>Send</Text></Pressable>
                    <Pressable onPress={()=>setModalVisible(false)} style={styles.modalPasswordResetButtonBack}><Text style={styles.modalPasswordResetButtonBackText}>Back</Text></Pressable>
                    </View>
                </View>
                </View>
            </Modal>
    );
}

export default ResetPasswordModal;