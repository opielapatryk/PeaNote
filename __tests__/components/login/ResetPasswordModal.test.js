import React from 'react';
import renderer from 'react-test-renderer';
import ResetPasswordModal from '../../../src/screens/login/components/ResetPasswordModal';

test('LoginWithApple button renders correctly.', () => {
    const tree = renderer.create(<ResetPasswordModal/>).toJSON();
    
    expect(tree).toMatchSnapshot();
});