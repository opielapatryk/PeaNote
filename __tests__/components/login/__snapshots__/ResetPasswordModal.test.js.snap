// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWithApple button renders correctly. 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 20,
          "elevation": 5,
          "margin": 20,
          "padding": 35,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "600",
            "marginBottom": 10,
          }
        }
      >
        Where should we send a reset link?
      </Text>
      <Text
        style={
          {
            "marginBottom": 10,
          }
        }
      >
        Enter the email associated with your account to reset your password.
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Email"
        style={
          {
            "borderColor": "lightgray",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginBottom": 25,
            "padding": 10,
          }
        }
        value=""
      />
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "black",
              "borderRadius": 5,
              "padding": 10,
              "width": 100,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontWeight": "600",
                "letterSpacing": 1,
                "textAlign": "center",
              }
            }
          >
            Send
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "borderWidth": 1,
              "padding": 10,
              "width": 100,
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "600",
                "letterSpacing": 1,
                "textAlign": "center",
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
